eta/core/datasets.py

```diff
@@ -744,7 +744,7 @@ def split(self, split_fractions=None, descriptions=None,
         dataset_copy = copy.deepcopy(self)
         dataset_list = []
         for dataset_index in dataset_indices:
-            dataset_copy.dataset_index = dataset_index
+            dataset_copy._dataset_index = dataset_index
```

Comment:
Better to create a setter method for `dataset_index`.

Reply:
Done!